src/mem_scan.c

```diff
@@ -131,6 +131,7 @@ void scan_mem(sqlite3 *db, int pid)
 		errx(1, "Cannot allocate %lu bytes for the kivp array container", sizeof(seen_kivp_wrapper_struct));
 	}
 	int initial_size = 100;
+	int growth_size = 100;
```

Comment:
I first doubling it each time the limit is reached, but then I thought it's not necessary to grow doubly each time because we are tracking the number of scanned vm block, I can't imagine it getting into the thousands?

The array holds a 100 of "struct kinit_vmentry" first, each kinit_vmentry struct contains about 125bytes (https://github.com/CTSRD-CHERI/cheribsd/blob/main/sys/sys/user.h#L554), so the array holds 12,500 bytes (~12Mb), doubling after three times it would make the array of size 800 of those vmentries and ~96Mb. If it grows linearly by adding a hundred each time, it would be ~49Mb after growing three times.

I guess with the expectation that it would not need to grow to very big, the memory consumption might not be a concern at all given the above calculations? 

So is the advice is better to double it for performance benefits?

Reply:
Once doubling gets large, it doesn't really commit memory in practice except in debug malloc configuration (it's address space and empty pages) copying is the dominant cost of realloc so it's generally best to do that.

Reply:
Thanks. I believe the last commit has fixed this then. Please review, thank you!